### PR TITLE
interval2maf remove from unversioned or deprecate

### PR DESCRIPTION
**Pull request by bernt-matthias**
In order to make the IUC version of the tool work (https://github.com/galaxyproject/tools-iuc/pull/4769) we need to

- remove it from the list of unversioned tools (b7ad595b7298729924a0744fba62ba7a6198410a)
- or just deprecate it (214d11c9214fc9a888db11bfc9aa186855a42fa4)

Intuitively I would suggest the later option since it is know to be broken and we have a working version at IUC (https://github.com/galaxyproject/tools-iuc/pull/4769 will be working after accepting this PR)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by mvdbeek** [comment]
Can you go with https://github.com/galaxyproject/galaxy/commit/b7ad595b7298729924a0744fba62ba7a6198410a ? I'm not sure it's wise to remove tools, just for the potential loss of tool state.

**Comment by nsoranzo** [comment]
If we don't remove it, maybe at least move it to a `deprecated/` subdirectory of tools, and remove it from the `lib/galaxy/config/sample/tool_conf.xml.sample` ? So people won't try to update it or add it to new Galaxy installations.

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -180,7 +180,6 @@
     "Interval_Maf_Merged_Fasta2",
     "GeneBed_Maf_Fasta2",
     "maf_stats1",
-    "Interval2Maf1",
```
Comment (mvdbeek) [inline lib/galaxy/tools/__init__.py:183]: Regardless of what you're doing in the end you need to move this to `GALAXY_LIB_TOOLS_VERSIONED`.

**Comment by bernt-matthias** [comment]
> Can you go with [b7ad595](https://github.com/galaxyproject/galaxy/commit/b7ad595b7298729924a0744fba62ba7a6198410a) ? I'm not sure it's wise to remove tools, just for the potential loss of tool state.

Will do. How about a comment in the tool: stating that no more updates should be done?

**Comment by nsoranzo** [comment]
> > Can you go with [b7ad595](https://github.com/galaxyproject/galaxy/commit/b7ad595b7298729924a0744fba62ba7a6198410a) ? I'm not sure it's wise to remove tools, just for the potential loss of tool state.
> 
> Will do. How about a comment in the tool: stating that no more updates should be done?

What about my suggestion? https://github.com/galaxyproject/galaxy/pull/14573#issuecomment-1253445618

**Comment by mvdbeek** [comment]
I think that if you've had the tool installed (and you've run it at leat once) you should keep it installed, so moving the location is also going to break any tool_conf.xml that pointed at the original location.

It's tricky ... if the IUC tool didn't change any of the parameter names maybe it does make sense to drop the file. I'll leave that decision up to you.

**Comment by bernt-matthias** [comment]
Btw. what exactly is "tool state"?

**Comment by nsoranzo** [comment]
> I think that if you've had the tool installed (and you've run it at least once) you should keep it installed, so moving the location is also going to break any `tool_conf.xml` that pointed at the original location.

Ah, you're right, should have thought about that! So I guess we are left with:
- From the second commit, keep only the changes to `lib/galaxy/config/sample/tool_conf.xml.sample` and `lib/galaxy/tool_util/xsd/galaxy.xsd`
- Add a comment to `tools/maf/interval2maf.xml` stating that the tool is deprecated in favour of the tools-iuc one.

> Btw. what exactly is "tool state"?

I guess Marius means the ability to re-run/inspect the metadata of old output datasets of this tool, plus not breaking old workflows.

**Comment by bernt-matthias** [comment]
Thanks for the precise suggestions @nsoranzo

**Pull request merged**